cip36/rust/src/error.rs

```diff
@@ -0,0 +1,10 @@
+#[derive(Debug, thiserror::Error)]
+pub enum CIP36Error {
+    #[error("Empty delegation array")]
+    EmptyDelegationArray,
+    // TODO: can we check this somehow against anything? I don't believe so, so maybe remove this
```

Comment:
TODO?

Reply:
This TODO is from what was already in `develop` in `lib.rs`. I can remove it if you want. In the CIP36 spec the stake address must be discriminated with the same network that the tx that contains the metadata is submitted it is on, but then I don't think there's a way to check this in this self-contained cip36 metadata library. It would have to involve extra data either passed in or done after the fact.